Makefile

```diff
@@ -99,26 +103,35 @@ XCFLAGS += $(CFLAGS) $(INC_FLAGS) -D UT_VERSION=\"$(VERSION)\"
 VPATH += $(UT_CORE_DIR)
 VPATH += $(TOP_DIR)
 
-.PHONY: clean list arm linux framework test createdirs
+.PHONY: clean list arm linux framework test createdirs all
+
+all: framework $(OBJS)
 
-all: framework test
+COMPLETION_MARKER := .download_and_build_done
```

Comment:
what is being completed? the framework?. Then you should put with the framework directory and not have it externally at the top level.

and it should be called what it stating

FRAMEWORK_IS_BUILT_MARKER = .framework_is_built